employees/tests/test_views.py

```diff
@@ -862,3 +863,70 @@ def test_view_should_display_author_reports_only_one_project(self):
         self.assertNotContains(response, report_from_other_project.description)
         self.assertEqual(response.status_code, 200)
         self.assertTemplateUsed(response, AuthorReportProjectView.template_name)
+
+
+class ReportSummaryTests(TestCase):
+    def setUp(self):
+        self.user = ManagerUserFactory()
+        self.current_time = timezone.now()
+        self.work_hours_per_project = self._fill_database_with_user_reports(self.user, self.current_time)
+        self.total_monthly_hours = sum(self.work_hours_per_project.values(), timezone.timedelta())
+
+    def test_report_list_should_accurately_evaluate_work_hours_statistics(self):
+        self.client.force_login(self.user)
+        url = reverse("custom-report-list", kwargs={"year": self.current_time.year, "month": self.current_time.month})
+
+        context_data = self.client.get(url).context_data
+        self._assert_projects_work_percentage_is_evaluated_accurately(context_data)
+
+    def test_author_report_list_should_accurately_evaluate_work_hours_statistics(self):
+        admin_user = AdminUserFactory()
+        self.client.force_login(admin_user)
+        url = reverse(
+            "author-report-list",
+            kwargs={"pk": self.user.pk, "year": self.current_time.year, "month": self.current_time.month},
+        )
+
+        context_data = self.client.get(url).context_data
+
+        self._assert_projects_work_percentage_is_evaluated_accurately(context_data)
+
+    def _assert_projects_work_percentage_is_evaluated_accurately(self, context_data):
+        self.assertTrue("projects_work_percentage" in context_data.keys())
+
+        projects_work_percentage = context_data["projects_work_percentage"]
+
+        expected_result = {
+            project: ([work_hours_sum, work_hours_sum / self.total_monthly_hours * 100])
+            for project, work_hours_sum in self.work_hours_per_project.items()
+        }
+
+        self.assertEqual(expected_result, projects_work_percentage)
+
+    @staticmethod
+    def _fill_database_with_user_reports(user, base_date):
```

Comment:
I consider this the primary function. The returned value is secondary. I believe I've seen something like that in C++ or Java libraries when some state-changing method returns some useful output in case you'd need it (can't name it, honestly I just feel like I've encountered something like this). I really needed this output for the rest of the test, but generating reports is a primary function and reason it is called.

I'm hesitant to change the name into some scary monster like **fill_database_with_user_reports_and_get_sum_of_all_work_hours_in_tested_month**.

Reply:
Ok, so on further inspection I came to a conclusion it would be appropriate to split these two functions.

Reply:
Not related to this PR anymore